src/IROperator.cpp

```diff
@@ -248,13 +273,62 @@ Expr const_false(int w) {
     return make_zero(UInt(1, w));
 }
 
+Expr lossless_cast(Type t, Expr e) {
+    if (t == e.type()) {
+        return e;
+    } else if (t.can_represent(e.type())) {
+        return cast(t, e);
+    }
+
+    if (const Cast *c = e.as<Cast>()) {
+        if (t == c->value.type()) {
```

Comment:
I think a comment here explaining that widening has already been handled by the can_represent case would help. (I'm worried about e.g. `cast<float>(some_double_not_exact_as_a_float)` or `cast<uint8_t>(1024)` causing this code to remove the precision reducing Cast and thus returning the wrong (full precision) value. If I understand, that can't happen because these cases were handled above.)